### PR TITLE
Add a link to restart to NCR cart page

### PR DESCRIPTION
**Pull request by cmc333333**
Also generalizes a CSS selector.

I don't like the ncr-specific check in the ruby code. It'd be much nicer if there was a way to say something like:

```
<%= render partial: cart.origin + '/restart_link', locals: {cart: cart} %>
```

and [account for errors](http://stackoverflow.com/questions/3559419/is-there-any-rails-function-to-check-if-a-partial-exists). Yea Nay? 

Looks like:

screen shot 2015-03-03 at 10 50 22 am

### CODE REVIEW

File: app/helpers/carts_helper.rb [hunk 1]
```diff
@@ -22,6 +22,11 @@ def display_response_actions?(cart, user)
     current_linear_approval?(cart, user)
   end
 
+  # Todo: Move this to an NCR specific template?
+  def display_restart?(cart, user)
```
Comment (afeld) [inline app/helpers/carts_helper.rb]: Minor: you can access `current_user` in here directly – don't need to pass it in. Unless you prefer passing it explicitly?

File: spec/features/ncr_proposal_spec.rb [hunk 2]
```diff
@@ -136,5 +136,36 @@
...
+      cart.setProp('origin', 'somewhereElse')
+      visit "/carts/#{cart.id}"
+      expect(page).not_to have_content('Restart this Cart?')
+
+      cart.setProp('origin', 'ncr')
+      req = cart.approvals.where(role: 'requester').first
+      req.update_attribute(:user, FactoryGirl.create(:user))
+      visit "/carts/#{cart.id}"
+      expect(page).not_to have_content('Restart this Cart?')
+    end
```
Comment (afeld) [inline spec/features/ncr_proposal_spec.rb]: Mind breaking these up into five separate tests?

**Comment by afeld** [comment]
> I don't like the ncr-specific check in the ruby code.

I think we will learn more as we build out the different use cases, but what you suggested sounds reasonable to me.

File: app/helpers/carts_helper.rb (hunk 1 above)
Comment (cmc333333) [inline app/helpers/carts_helper.rb]: Booo global state. I'll rewrite it, though ;)

Comment (afeld) [inline app/helpers/carts_helper.rb]: _Technically_ it's another method (that has internal state), but I hear ya. Whatever you prefer!

**Comment by cmc333333** [comment]
Created a `client_partial` function, which will include a partial template if the template is present in the client/origin directory.

Also cleaned up some redundancy in the tests related to restarting for ncr.

File: app/models/cart.rb [hunk 3]
```diff
@@ -203,7 +203,9 @@ def import_cart_items(cart_items_params)
   end
 
   def origin
-    self.getProp('origin')
+    # In practice, carts should always have an origin. Account for test cases
+    # and old data with this "or"
+    self.getProp('origin') || ''
```
Comment (afeld) [inline app/models/cart.rb]: TODO backfill `origin`s on old Carts.

**Pull request merged**